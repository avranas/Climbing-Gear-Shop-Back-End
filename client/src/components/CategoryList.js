import { Link } from 'react-router-dom';
import shoesPic from '../images/shoes.png';
import harnessPic from '../images/harness.png';
import ropePic from '../images/rope.png';
import carabinerPic from '../images/carabiner.png';
import chalkPic from '../images/chalk.png';



const CategoryList = (props) => {

  const linkStyle = {
    color: 'black',
    textDecoration: 'none'
  };

  return (
    <div id="category-list" className="d-flex justify-content-around flex-wrap">
      <Link to="/products?category=shoes" style={linkStyle}>
        <div className="category-box">
          <img src={shoesPic} alt="shoes"/>
          <p>Shoes</p>
        </div>
      </Link>
      <Link to="/products?category=harnesses" style={linkStyle}>
        <div className="category-box">
          <img src={harnessPic} alt="harnesses" />
          <p>Harnesses</p>
        </div>
      </Link>
      <Link to="/products?category=ropes" style={linkStyle}>
        <div className="category-box">
          <img src={ropePic} alt="ropes"/>
          <p>Ropes</p>
        </div>
      </Link>
      <Link to="/products?category=hardware" style={linkStyle}>
        <div className="category-box">
          <img src={carabinerPic} alt="hardware"/>
          <p>Hardware</p>
        </div>
      </Link>
      <Link to="/products?category=chalk" style={linkStyle}>
        <div className="category-box">
          <img src={chalkPic} alt="chalk"/>
          <p>Chalk</p>
        </div>
      </Link>
    </div>
  );
};

export default CategoryList;

