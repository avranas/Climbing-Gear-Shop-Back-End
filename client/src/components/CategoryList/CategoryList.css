
#category-list {
  display: flex;
}

.category-box {
  margin: 1rem;
}
.category-box p {
  text-align: center;
  font-size: 1.5rem;
  font-weight: 600;
  padding-bottom: 1rem;
}

.category-box img {
  width: 14rem;
  height: 14rem;
  padding: 2rem;
  margin-bottom: 0;
}
.category-box p {
  margin-top: 0;
}
.category-box:hover {
  text-decoration: underline;
  
}
