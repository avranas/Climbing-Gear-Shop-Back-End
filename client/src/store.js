import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});