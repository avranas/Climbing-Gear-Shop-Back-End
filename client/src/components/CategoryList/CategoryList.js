import { Link } from "react-router-dom";
import shoesPic from "../../images/shoes.png";
import harnessPic from "../../images/harness.png";
import ropePic from "../../images/rope.png";
import carabinerPic from "../../images/carabiner.png";
import chalkPic from "../../images/chalk.png";
import "./CategoryList.css";

const CategoryList = (props) => {
  const linkStyle = {
    color: "rgb(48, 53, 73)",
    textDecoration: "none",
  };

  return (
    <section
      id="category-list"
      className="container d-flex justify-content-around flex-wrap"
    >
      <Link to="/products/0?category=shoes" style={linkStyle}>
        <div className="category-box styled-box styled-link-box">
          <img src={shoesPic} alt="shoes" />
          <p>Shoes</p>
        </div>
      </Link>
      <Link to="/products/0?category=harnesses" style={linkStyle}>
        <div className="category-box styled-box styled-link-box">
          <img src={harnessPic} alt="harnesses" />
          <p>Harnesses</p>
        </div>
      </Link>
      <Link to="/products/0?category=ropes" style={linkStyle}>
        <div className="category-box styled-box styled-link-box">
          <img src={ropePic} alt="ropes" />
          <p>Ropes</p>
        </div>
      </Link>
      <Link to="/products/0?category=hardware" style={linkStyle}>
        <div className="category-box styled-box styled-link-box">
          <img src={carabinerPic} alt="hardware" />
          <p>Hardware</p>
        </div>
      </Link>
      <Link to="/products/0?category=chalk" style={linkStyle}>
        <div className="category-box styled-box styled-link-box">
          <img src={chalkPic} alt="chalk" />
          <p>Chalk</p>
        </div>
      </Link>
    </section>
  );
};

export default CategoryList;
