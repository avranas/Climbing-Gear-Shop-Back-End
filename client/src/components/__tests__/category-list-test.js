import { cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";
import CategoryList from "../CategoryList/CategoryList";

afterEach(() => {
  cleanup();
});

const categoryList = (
  <Provider store={store}>
    <BrowserRouter>
      <CategoryList />
    </BrowserRouter>
  </Provider>
);

test("CategoryList matches snapshot", async () => {
  const tree = renderer.create(categoryList).toJSON();
  expect(tree).toMatchSnapshot();
});
