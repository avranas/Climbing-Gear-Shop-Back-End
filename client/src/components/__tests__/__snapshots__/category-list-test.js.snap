// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList matches snapshot 1`] = `
<section
  className="container d-flex justify-content-around flex-wrap"
  id="category-list"
>
  <a
    href="/products/0?category=shoes"
    onClick={[Function]}
    style={
      Object {
        "color": "rgb(48, 53, 73)",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="category-box styled-box styled-link-box"
    >
      <img
        alt="shoes"
        src="shoes.png"
      />
      <p>
        Shoes
      </p>
    </div>
  </a>
  <a
    href="/products/0?category=harnesses"
    onClick={[Function]}
    style={
      Object {
        "color": "rgb(48, 53, 73)",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="category-box styled-box styled-link-box"
    >
      <img
        alt="harnesses"
        src="harness.png"
      />
      <p>
        Harnesses
      </p>
    </div>
  </a>
  <a
    href="/products/0?category=ropes"
    onClick={[Function]}
    style={
      Object {
        "color": "rgb(48, 53, 73)",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="category-box styled-box styled-link-box"
    >
      <img
        alt="ropes"
        src="rope.png"
      />
      <p>
        Ropes
      </p>
    </div>
  </a>
  <a
    href="/products/0?category=hardware"
    onClick={[Function]}
    style={
      Object {
        "color": "rgb(48, 53, 73)",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="category-box styled-box styled-link-box"
    >
      <img
        alt="hardware"
        src="carabiner.png"
      />
      <p>
        Hardware
      </p>
    </div>
  </a>
  <a
    href="/products/0?category=chalk"
    onClick={[Function]}
    style={
      Object {
        "color": "rgb(48, 53, 73)",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="category-box styled-box styled-link-box"
    >
      <img
        alt="chalk"
        src="chalk.png"
      />
      <p>
        Chalk
      </p>
    </div>
  </a>
</section>
`;
